lib/jorb/backend.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Jorb.Backend do
+  @moduledoc ~S"""
+  The backend behavior represents a queueing abstraction.
+
+  Jorb requires four functions to be defined for a backend:
+  `c:setup/1`, `c:enqueue/2`, `c:dequeue/1`, and `c:finalize/2`
+
+
+  Setup handles anything to create the underlying queue
+  Enqueue pushes a message on to the queue
+  Dequeue pulls the next message from the queue
+  Finalize removes the given message from the queue
+  """
+
+  @callback setup(String.t()) :: :ok | no_return
+  @callback enqueue(String.t(), Map.t()) :: {:ok, Map.t()} | {:error, any}
+  @callback dequeue(String.t()) :: [Map.t()]
```

Comment:
I still don't know why you're mixing tagged tuples and exceptions in the same interface.  This is a lot more important to get right than `Jorb.Backends.File` because you can't change it without breaking userspace.  I have a slight preference toward tagged tuples, but I feel pretty strongly that it should at least be consistent.

Reply:
Thinking more, I like tagged tuples for now because you can always add bang methods later.  Going the other way, you'd have to break the API.

Reply:
The only thing that exceptions is `setup`, which I think should exception. If a queue can't be set up, the program should crash.

`dequeue` or whatever I rename it to, returns `[]` if there's nothing to return

Reply:
I'm not talking about "nothing to return", I'm talking about stuff like "broken pipe".